apps/src/applab/dataBrowser/DataTable.jsx

```diff
@@ -29,6 +41,75 @@ const DataTable = React.createClass({
     onViewChange: React.PropTypes.func.isRequired
   },
 
+  getInitialState() {
+    return {
+      newColumns: [],
+      editingColumn: null
+    };
+  },
+
+  componentWillReceiveProps(nextProps) {
+    // Forget about new columns when switching between tables.
+    if (this.props.tableName !== nextProps.tableName) {
+      this.setState({newColumns: []});
```

Comment:
Should you also forget about the column being edited when switching between tables?